website/docs/d/iam_policy_document.html.markdown

```diff
@@ -296,4 +202,94 @@ data "aws_iam_policy_document" "override_json_example" {
 }
 ```
 
-You can also combine `source_json` and `override_json` in the same document.
+## Argument Reference
+
+The following arguments are supported:
+
+* `policy_id` (Optional) - An ID for the policy document.
+* `source_json` (Optional) - An IAM policy document to import as a base for the
+  current policy document.  Statements with non-blank `sid`s in the current
+  policy document will overwrite statements with the same `sid` in the source
+  json.  Statements without an `sid` cannot be overwritten. Can be combined with
+  `override_json`.
+* `override_json` (Optional) - An IAM policy document to import and override the
+  current policy document.  Statements with non-blank `sid`s in the override
+  document will overwrite statements with the same `sid` in the current document.
+  Statements without an `sid` cannot be overwritten. Can be combined with
+  `source_json`.
+* `statement` (Required) - A nested configuration block (described below)
```

Comment:
It would be very helpful to make `statement` optional so that policies can be layered more flexibly. Even if `statement` policy docs provide the foundational level, when layering policies, you need the capability to just compose policies without needing a new `statement`. To get around this, you have to specify a dummy `statement` that will be purposely overridden, when all you really want is the `source_json` and possibly `override_json`. See PR #6052.